GVFS/GVFS.Virtualization/Background/FileSystemTask.cs

```diff
@@ -27,7 +27,7 @@ public enum OperationType
             OnFolderRenamed,
             OnFolderDeleted,
             OnFolderFirstWrite,
-            OnIndexWriteWithoutProjectionChange,
+            OnIndexWriteRequiringModifiedPathsValidation,

```

Comment:
Did the meaning of this event change? Or is this just a rename to make it more descriptive? (I'm wondering if we need an upgrade step for this)

Reply:
Are these serialized as `int` values or `string`?  I always forget.  If it is as strings then we will hit the default case in the switch.
```
default:
                     throw new InvalidOperationException("Invalid background operation");	                     throw new InvalidOperationException("Invalid background operation");
```

Reply:
Fortunately these are serialized as ints:

*Example*

```
E:\os6\.gvfs\databases>type BackgroundGitOperations.dat
A 3 6 ReleaseConfig.dc
A 4 1 foofoo
```